brave/src/main/java/brave/Tracer.java

```diff
@@ -44,7 +44,35 @@
  * @see Span
  * @see Propagation
  */
-public final class Tracer {
+public final class Tracer implements Closeable {
+  // volatile for visibility on get. writes guarded by Tracer.class
+  static volatile Tracer current = null;
+
+  /**
+   * Returns the most recent tracer that hasn't been closed.
+   *
+   * <p>This object should not be cached.
+   */
+  @Nullable public static Tracer current() {
```

Comment:
All very good points which illustrator the simplicity of your current approach.  Given that current use cases are already calling `Tracer.current()` for each execution and that this pattern is discourage, I think it makes a sense to keep it as is.  Thanks for the thorough reply :)